src/tests/ast_utils_test.cpp

```diff
@@ -334,5 +334,44 @@ TEST(AstUtils, ReorderClauseAtoms) {
             toString(*reorderedClause1));
 }
 
+TEST(AstUtils, RemoveClauseRedundancies) {
```

Comment:
This could do with a teensy bit of documentation to describe the behaviour that you're testing.